sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -191,25 +199,23 @@ internal static (string FormattedError, string? ErrorCode, IDictionary<string, s
             return (formatMessage, error?.Code, data);
         }
 
-        /// <summary>
-        /// This is intentionally sync-only as it will only be called by the ctor.
-        /// </summary>
-        /// <param name="response"></param>
-        /// <returns></returns>
-        internal static string? ReadContent(Response response)
+        private static void BufferResponseIfNeeded(Response response)
         {
-            string? content = null;
-
-            if (response.ContentStream != null &&
-                ContentTypeUtilities.TryGetTextEncoding(response.Headers.ContentType, out var encoding))
+            // Buffer into a memory stream if not already buffered
+            if (response.ContentStream is null or MemoryStream)
```

Comment:
We ran into an issue with checking `is MemoryStream` related to [this issue](https://github.com/Azure/azure-sdk-for-net/issues/35355), where if you use `BinaryData.FromStream()` to buffer the stream, the `is MemoryStream` check will fail.

Is `is MemoryStream` a stand-in check for indicating whether we've buffered the stream in the buffering policy?  If we buffered it elsewhere differently, is that valid?  It would be good to understand the intention here a little better, I think.

Reply:
> Is there any advantage to checking via is MemoryStream vs CanSeek? I think we should be consistent unless there is a clear advantage.

We are consistent now because I removed the other CanSeek check.

> We ran into an issue with checking is MemoryStream related to https://github.com/Azure/azure-sdk-for-net/issues/35355, where if you use BinaryData.FromStream() to buffer the stream, the is MemoryStream check will fail.

This shouldn't be possible anymore because the stream will be buffered before the DetailParser method is invoked.

> Is is MemoryStream a stand-in check for indicating whether we've buffered the stream in the buffering policy? If we buffered it elsewhere differently, is that valid? It would be good to understand the intention here a little better, I think.

Yes, it is checking whether the stream has already been buffered. If someone injects a custom policy and does something to the content, as long as we still have a MemoryStream we consider it buffered.

Reply:
> We are consistent now

We still use `CanSeek` in `ResponseBodyPolicy`

Reply:
But that is specifically because we are seeking there. We have no such requirement here.